libfreerdp/core/window.c

```diff
@@ -86,8 +86,8 @@ BOOL update_read_icon_info(wStream* s, ICON_INFO* iconInfo)
 	Stream_Read_UINT16(s, iconInfo->width); /* width (2 bytes) */
 	Stream_Read_UINT16(s, iconInfo->height); /* height (2 bytes) */
 
-	/* cbColorTable is only present when bpp is 1, 2 or 4 */
-	if (iconInfo->bpp == 1 || iconInfo->bpp == 2 || iconInfo->bpp == 4)
```

Comment:
Yes. It's mentioned in the commit message (so not immediately visible in the diff).

[MS-RDPERP 2.2.1.2.3 Icon Info (TS_ICON_INFO)](https://msdn.microsoft.com/en-us/library/cc242447.aspx):

> **CbColorTable (2 bytes):** An unsigned 16-bit integer. The size, in bytes, of the color table data. This field is ONLY present if the **bits per pixel (Bpp)** value is 1, 4, or 8.

_Technically_, DIB format uses palettes for 1, 2, 4, 8 bpp formats, but RAIL seems to support only 1, 4, 8 bpp. I've never encountered 1 and 4 bpp icons, but 8 bpp were causing troubles because of this bug, the alpha and color tables were incorrectly parsed and the icon could not be processed.